### PR TITLE
Add GH Actions Pipeline

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

Continuation of #62759 from the salt repository instead of a fork

### CODE REVIEW

**Review by twangboy — DISMISSED** [review]

**Review by whytewolf — DISMISSED** [review]

File: .github/workflows/test-action-macos.yml [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+          name: testrun-artifacts-${{ inputs.distro-slug }}-${{ env.NOX_SESSION }}
+          path: artifacts
+
+#      - name: Publish Test Report
+#        uses: mikepenz/action-junit-report@v3
+#        if: always() && steps.download-artifacts-from-vm.outcome == 'success'
+#        with:
+#          check_name: Test Results(${{ inputs.distro-slug }} ${{ matrix.tests-chunk }})
+#          report_paths: 'artifacts/xml-unittests-output/*.xml'
+
```
Comment (MKLeb) [inline .github/workflows/test-action-macos.yml:247]: Leftover or still needed?

File: .github/workflows/test-action.yml [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+        with:
+          name: testrun-artifacts-${{ inputs.distro-slug }}-${{ env.NOX_SESSION }}
+          path: artifacts
+
+#      - name: Publish Test Report
+#        uses: mikepenz/action-junit-report@v3
+#        if: always() && steps.download-artifacts-from-vm.outcome == 'success'
+#        with:
+#          check_name: Test Results(${{ inputs.distro-slug }} ${{ matrix.tests-chunk }})
+#          report_paths: 'artifacts/xml-unittests-output/*.xml'
```
Comment (MKLeb) [inline .github/workflows/test-action.yml:254]: Same question as before

File: .github/workflows/test-action-macos.yml (hunk 1 above)
Comment (s0undt3ch) [reply .github/workflows/test-action-macos.yml:247]: Will be needed in the near future

File: .github/workflows/test-action.yml (hunk 2 above)
Comment (s0undt3ch) [reply .github/workflows/test-action.yml:254]: Same as above

File: tests/conftest.py [hunk 3]
```diff
@@ -277,6 +277,14 @@ def pytest_configure(config):
...
         ),
     )
+    config.addinivalue_line(
+        "markers",
+        "skip_initial_gh_actions_failure(skip=<boolean or callable, reason=None): Skip known test failures "
+        "under the new GH Actions setup if the environment variable SKIP_INITIAL_GH_ACTIONS_FAILURES "
+        "is equal to '1' and the 'skip' keyword argument is either `True` or it's a callable that "
+        "when called returns `True`. If `skip` is a callable, it should accept a single argument "
+        "'grains', which is the grains dictionary.",
+    )
```
Comment (MKLeb) [inline tests/conftest.py:287]: This is for your initial phase, yeah?  Ideally we get all these fixed in the future?

**Review by MKLeb — DISMISSED** [review]

File: tests/conftest.py (hunk 3 above)
Comment (s0undt3ch) [reply tests/conftest.py:287]: Yeah, this marker is setup to set it to false and see what needs fixing.

**Review by Ch3LL — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by whytewolf — DISMISSED** [review]

**Review by Ch3LL — DISMISSED** [review]

**Review by Ch3LL — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Review by MKLeb — APPROVED** [review]

**Pull request merged**